src/components/CustomTable/CustomTable.tsx

```diff
@@ -0,0 +1,254 @@
+        if (error instanceof FetchError) {
+          throw new FetchError(`Failed to fetch with error: ${error.message}`);
+        }
+      });
+  });
+
+  const renderRow = (row: T, rowIdx: number) => {
+    const isItemSelected = isSelected(row);
+    const labelId = `enhanced-table-checkbox-${rowIdx}`;
+
+    return (
+      <TableRow
+        hover
+        onClick={(event: MouseEvent<unknown>) => {
+          return handleClick(event, row);
+        }}
+        role='checkbox'
+        aria-checked={isItemSelected}
+        tabIndex={-1}
+        key={row.id as number}
+        selected={isItemSelected}
+      >
+        <TableCell padding='checkbox'>
+          <Checkbox
+            color='primary'
+            checked={isItemSelected}
+            inputProps={{
+              'aria-labelledby': labelId,
+            }}
+          />
+        </TableCell>
+        {Object.entries(row).map(([key, value]) => {
+          return (
+            <TableCell
+              key={key}
+              align='left'
+            >
+              {String(value)}
+            </TableCell>
+          );
+        })}
+      </TableRow>
+    );
+  };
+
+  const renderHead = () => {
+    const label = t(`${tableName}.selectAll`);
+    return (
+      <TableHead>
+        <TableRow>
+          <TableCell padding='checkbox'>
+            <Checkbox
+              color='primary'
+              indeterminate={selectedRowsIds.length > 0 && selectedRowsIds.length < rows.length}
+              checked={rows.length > 0 && selectedRowsIds.length === rows.length}
+              onChange={handleSelectAllClick}
+              inputProps={{
+                'aria-label': label,
+              }}
+            />
+          </TableCell>
+          {headerCells.map((headCell, index) => {
+            return (
+              <TableCell
+                key={String(headCell.id)}
+                align='left'
+                sortDirection={sortingConfig[index].columnName === String(headCell.id) ? sortingConfig[0].sortDirection : false}
+              >
+                <TableSortLabel
+                  active={sortingConfig[index].columnName === String(headCell.id)}
+                  direction={sortingConfig[index].columnName === String(headCell.id) ? sortingConfig[0].sortDirection : SortingDirection.ASC}
+                  onClick={createSortHandler(headCell.id)}
+                >
+                  {headCell.label}
+                  {sortingConfig[index].columnName === headCell.id ? (
+                    <Box
+                      component='span'
+                      sx={{ display: 'none' }}
+                    >
+                      {sortingConfig[index].sortDirection === SortingDirection.DESC ? t(`${tableName}.sortDsc`) : t(`${tableName}.sortAsc`)}
+                    </Box>
+                  ) : null}
+                </TableSortLabel>
+              </TableCell>
+            );
+          })}
+        </TableRow>
+      </TableHead>
+    );
+  };
```

Comment:
I think this can be a separated component

Reply:
fixed in [0fc627d](https://github.com/dorm-social-room-manager/dsrm-frontend/pull/38/commits/0fc627d60e2e53a288baba4ce60920d906badc07)